Csocket.cc

```diff
@@ -2691,39 +2746,56 @@ CS_STRING Csock::GetPeerPubKey() const
 {
 	CS_STRING sKey;
 
-	SSL_SESSION * pSession = GetSSLSession();
+	X509 * pCert = GetX509();
```

Comment:
The old code with GetSSLSession calls [SSL_get_session](https://github.com/openssl/openssl/blob/a230b26e0959dc5f072fbbdadcc9ed45e904c50c/ssl/ssl_sess.c#L47-L51) which returns `ssl->session`. Later the code accesses the `peer` property of that object.

GetX509 calls [SSL_get_peer_certificate](https://github.com/openssl/openssl/blob/cfd20f64cc4bd440cfc8fe59f2daaa575015af3d/ssl/ssl_lib.c#L1322-L1337) which you can see directly returns `ssl->session->peer` (bypassing the now opaque session object).

The only meaningful difference here is that now an internal reference count is incremented on the cert, hence it needs to be freed after use.